Add Snabb Switch package

lukego opened the pull request:
Add [Snabb Switch](https://github.com/SnabbCo/snabbswitch) as `tools.networking.snabb`.

Add me to the maintainers file as `lukego` too (this is my first contribution).

lukego committed:
maintainers.nix: Add lukego

lukego committed:
snabb: Added Snabb Switch (v2015.10) in tools/networking

I have tested this manually with nix-env.

domenkozar commented on pkgs/tools/networking/snabb/default.nix:
```diff
@@ -0,0 +1,25 @@
+  name = "snabb-v2015.10";
+
+  src = fetchgit {
+    url = "git://github.com/SnabbCo/snabbswitch";
```
Best to use https since git:// my be blocked by some enterprise firewalls

domenkozar commented on pkgs/tools/networking/snabb/default.nix:
```diff
@@ -0,0 +1,25 @@
+
+  src = fetchgit {
+    url = "git://github.com/SnabbCo/snabbswitch";
+    rev = "20532eff595fef4bf1575d70e7963060a91add45";
```
Actually best to use https://github.com/SnabbCo/snabbswitch/archive/v2015.10.tar.gz (we prefer releases over git revisions)

jagajaga commented on pkgs/tools/networking/snabb/default.nix:
```diff
@@ -0,0 +1,24 @@
+
+meta = with stdenv.lib; {
+    homepage = "https://github.com/SnabbCo/snabbswitch";
+    description = "Snabb Switch: simple and fast packet networking toolkit";
```
Please omit pkg's name in the description.

jagajaga commented:
Please squash commits and don't forget about our naming guide :)

lukego commented:
Thanks for the review gents.

Initially I followed the instructions in [Contributing to nixpkgs](https://nixos.org/wiki/Contributing_to_nixpkgs) but it seems like I should instead follow [Nixpkgs contributors guide](http://nixos.org/nixpkgs/manual/) to get tips like squashing fixup commits, writing commit messages, and choosing package/version names.

Have now:
1. Squashed the fixup commits.
2. Fixed the version name (`snabb-2015.10` vs `snabb-v2015.10`).
3. Added a `longDescription` attribute.

edolstra commented on pkgs/tools/networking/snabb/default.nix:
```diff
@@ -0,0 +1,32 @@
+stdenv.mkDerivation rec {
+  name = "snabb-2015.10";
+
+  src = fetchurl {
```
This should use `fetchFromGitHib`. `fetchurl` hashes tend to break with GitHub (since the tarballs are generated dynamically).

lukego replied:
@edolstra Thanks for the tip! I will look into this on the next update.
